cmd/neofs-node/config.go

```diff
@@ -790,6 +791,9 @@ func initObjectPool(cfg *config.Config) (pool cfgObjectRoutines) {
 	pool.putRemote, err = ants.NewPool(pool.putRemoteCapacity, optNonBlocking)
 	fatalOnErr(err)
 
+	pool.putLocal, err = ants.NewPool(-1) // -1 stands for an infinite capacity worker
```

Comment:
did u test this change? does it affect anything? afaik https://pkg.go.dev/github.com/nspcc-dev/neofs-node/pkg/util#NewPseudoWorkerPool used by default also provides an infinite pool

Reply:
let's sync the terms. consider the [example](https://gist.github.com/carpawell/fd443d752aee1961d11efd9c3ef05894) and its output:
```
ants took: 10ms
sync pool took: 1038ms
```

that is what i (and `ants`'s creator) call an infinite pool: if you have a cycle that `Submit`s some task an infinite loop won't lock any iteration ever (in other words `numOfJobs` in the example can be any number and every job will always have a separate routine). on the other hand, `PseudoWorkerPool` just calls a func and always blocks

> if you have a cycle that `Submit`s some task

and that is what [we have](https://github.com/nspcc-dev/neofs-node/blob/c817194269943e4ed56d85eedf33b529bb1b282e/pkg/services/object/put/distributed.go#L180-L224) in our distribution logic: every local put task is done syncly (even if we have more remote node to handle)

Reply:
> used by default also provides an infinite pool

so answering this part: `PseudoWorkerPool` does not provide a pool by design (finite or infinite), it always blocks. not smth we expect when trying to parallelize some i/o operations

Reply:
got it now, thx